### ISSUE TITLE
Maximum call stack size exceeded error

### ISSUE THREAD

**Issue by Raphcal** [kr1sp1n/node-vault#179]
The following code results in a `Maximum call stack size exceeded`:

~~~javascript
var newVaultClient = require('node-vault');

function doSomethingUsingASecret(path) {
  var vaultClient = newVaultClient();
  var data = vaultClient.read(path);
  // Do something with data
}

for (var index = 0; index < 20; index++) {
  doSomethingUsingASecret("somepath");
}
~~~

With the following stack trace:

~~~
(node:1) UnhandledPromiseRejectionWarning: RangeError: Maximum call stack size exceeded
at String.toString (<anonymous>)
at isPlainObject (/app/node_modules/extend/index.js:17:20)
at extend (/app/node_modules/extend/index.js:95:27)
at extend (/app/node_modules/extend/index.js:104:51)
at /app/node_modules/request/index.js:88:5
at /app/node_modules/request/index.js:100:12
at /app/node_modules/request/index.js:100:12
at /app/node_modules/request/index.js:100:12
at /app/node_modules/request/index.js:100:12
at /app/node_modules/request/index.js:100:12
at /app/node_modules/request/index.js:100:12
at /app/node_modules/request/index.js:100:12
at /app/node_modules/request/index.js:100:12
at /app/node_modules/request/index.js:100:12
at /app/node_modules/request/index.js:100:12
~~~

Because `request` is wrapped by `defaults` each time a new instance of `vaultClient` is created:

~~~javascript
rp = (config['request-promise'] || rp).defaults(rpDefaults);
~~~

**Comment by bfaulk96** [comment]
@kr1sp1n I'm sure you're busy, especially since this is an Open Source project, but is there any chance you could take a look at @Raphcal 's PR to fix this? It's a pretty big bug, and there isn't much workaround other than forking your repo and using the fork, which isn't ideal.

Thanks for your hard work and contributions to this library 😄

**Comment by kr1sp1n** [comment]
@bfaulk96 PR released as 0.9.22

**Comment by bfaulk96** [comment]
You the real MVP. Thank you!
